### PR TITLE
CVE-2014-1744 and CVE-2013-2844

### PR DESCRIPTION
**Pull request by sxHelloWorld**
Integer Overflow and Use after free vulnerabilities

### CODE REVIEW

File: cves/CVE-2013-2844.yml [hunk 1]
```diff
@@ -32,7 +32,9 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
```
Comment (bjpicciano) [inline cves/CVE-2013-2844.yml]: More detail in the description on how this happened in the css engine would be good.

**Comment by sxHelloWorld** [comment]
The description is fixed.

File: cves/CVE-2014-1744.yml [hunk 2]
```diff
@@ -32,7 +32,9 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Integer overflow in the audio component occurs when a remote attacker sends an
+  overflow value to cause a denial of service or a large shared-memory allocation.
```
Comment (kbaumzie) [inline cves/CVE-2014-1744.yml]: This should be about two paragraphs and should be different from what I can find on the cve website

File: cves/CVE-2013-2844.yml [hunk 3]
```diff
@@ -32,7 +32,12 @@ description_instructions: |
...
+  Use after free vulnerability occurs when the CSS renderer attempts to use
+  the same pointer after freed causes chrome to crash. Hacker can use style
+  resolution as vulnerability access to cause an unspecific impact. This is
+  done by placing a CSS file on the server and wait for a client to fall in
+  the bait.
```
[the remark this answers is not in the document]
Comment (sxHelloWorld) [reply cves/CVE-2013-2844.yml]: I think I have provided sufficient description for this vulnerability. Can you elaborate on what I should put down?

File: cves/CVE-2014-1744.yml [hunk 4]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: CWE-680
+CWE: 680
```
Comment (ZacharyMoran) [inline cves/CVE-2014-1744.yml]: I can see the potential of this issue being CWE-680, but without doing the research myself your descriptions point more towards a simple Integer Overflow or Wraparound (CWE-190). Are you sure the vulnerability was leading to that integer being used to allocate memory and as such resulting in a Buffer Overflow? If so then keep the above, if this is not the case I suggest making the swap over to CWE-190.

File: cves/CVE-2014-1744.yml (hunk 2 above)
Comment (ZacharyMoran) [reply cves/CVE-2014-1744.yml]: Again, I am going to disagree. I believe the developer here has provided enough of a distinction between the given description and the CVE website. I do think there could be a bit of information added to express how an audio file can even have an integer overflow value that can be registered in the system as I am not entirely clear on how this occurs.

File: cves/CVE-2014-1744.yml (hunk 4 above)
Comment (ZacharyMoran) [reply cves/CVE-2014-1744.yml]: I pull back my above statements, you have the correct CWE however there should be mention of the Buffer Overflow part in your description as that was what I was using as my reasoning behind the original comment

File: cves/CVE-2013-2844.yml [hunk 5]
```diff
@@ -32,7 +32,13 @@ description_instructions: |
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Use after free vulnerability occurs when the CSS renderer attempts to use
+  the same pointer after freed causes chrome to crash. An attacker can use
+  style resolution as vulnerability access to cause an unspecific impact.
+  This is done by placing a CSS file on the server and wait for a client to
+  fall in the bait. As a consequence, memory might be accessed by something
+  that is not supposed to happen or data on Chrome might be lost.
```
Comment (bjpicciano) [inline cves/CVE-2013-2844.yml]: May be good to note some of the vulnerabilities caused by this (Denial of service could be one)

File: cves/CVE-2013-2844.yml [hunk 6]
```diff
@@ -165,8 +183,11 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: Due to the ability of external networks to free the CSS renderer,
+    the memory pointers may be freed and reused again outside of the control
+    of the Chrome application. This vulnerability can lead to potential crashes
+    or denial of service vulnerabilities.
```
Comment (mesternefeld) [inline cves/CVE-2013-2844.yml]: Please indent these lines so the CI passes.
```suggestion
    note: |
      Due to the ability of external networks to free the CSS renderer,
      the memory pointers may be freed and reused again outside of the control
      of the Chrome application. This vulnerability can lead to potential crashes
      or denial of service vulnerabilities.
```

File: cves/CVE-2013-2844.yml [hunk 7]
```diff
@@ -165,8 +183,11 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: Due to the ability of external networks to free the CSS renderer,
```
Comment (mesternefeld) [inline cves/CVE-2013-2844.yml]: Please indent these lines so the CI passes.
```suggestion
    note: |
      Due to the ability of external networks to free the CSS renderer,
      the memory pointers may be freed and reused again outside of the control
      of the Chrome application. This vulnerability can lead to potential crashes
      or denial of service vulnerabilities.
```

File: cves/CVE-2013-2844.yml [hunk 8]
```diff
@@ -165,8 +183,15 @@ lessons:
...
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Due to the ability of external networks to free the CSS renderer,
+      the memory pointers may be freed and reused again outside of the control
+      of the Chrome application. This vulnerability can lead to potential crashes
+      or denial of service vulnerabilities.
+    the memory pointers may be freed and reused again outside of the control
```
Comment (mesternefeld) [inline cves/CVE-2013-2844.yml]: Please indent the rest of the note so it follows the one above it to fix the CI

File: cves/CVE-2014-1744.yml (hunk 4 above)
Comment (bjpicciano) [inline cves/CVE-2014-1744.yml]: +2 upvotes

File: cves/CVE-2013-2844.yml [hunk 9]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: 416
```
Comment (bjpicciano) [inline cves/CVE-2013-2844.yml]: +4 upvotes

**Pull request merged**